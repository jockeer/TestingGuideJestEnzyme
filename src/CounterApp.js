import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({value = 10}) => {

    const [ counter, setCounter ] = useState(value)

    const sumar = (e) => {
        setCounter(counter+1)
    }
    const reset = (e) => {
        setCounter(value)
    }
    const restar = (e) => {
        setCounter(counter-1)
    }

    return ( 
        <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>

        <button onClick={ sumar } >+1</button>
        <button onClick={ reset } >reset</button>
        <button onClick={ restar } >-1</button>
        </>
     );
}

CounterApp.propTypes = {
    value: PropTypes.number
}
 
export default CounterApp;