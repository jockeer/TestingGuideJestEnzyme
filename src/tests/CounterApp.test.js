import { shallow } from 'enzyme'
import CounterApp from '../CounterApp'

describe('Prueba <CounterApp />', () => {

    let wrapper = shallow(<CounterApp />)

    beforeEach(()=>{
        wrapper = shallow(<CounterApp />)
    })

    test('debe mostrar <CounterApp /> correctamente con sus valores por defecto', () => {
        
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar el valor por defecto de 100', () => {
        
        const valor = 100
        const wrapper = shallow(<CounterApp value={valor}/>)
        const textoContador =  parseInt(wrapper.find('h2').text().trim()) 
        expect(textoContador).toBe(valor)
        

    })

    test('Debe incrementar con el boton +1', () => {

        wrapper.find('button').at(0).simulate('click');
        const textoContador =  parseInt(wrapper.find('h2').text().trim())
        expect(textoContador).toBe(11)

    })

    test('Debe decrementar con el boton -1', () => {

        wrapper.find('button').at(2).simulate('click');
        const textoContador =  parseInt(wrapper.find('h2').text().trim())
        expect(textoContador).toBe(9)

    })

    test('Debe decrementar con el boton -1', () => {

        wrapper.find('button').at(2).simulate('click');
        const textoContador =  parseInt(wrapper.find('h2').text().trim())
        expect(textoContador).toBe(9)

    })

    test('Debe colocar el valor por defecto con el boton reset', () => {
        const valor = 105
        const wrapper = shallow(<CounterApp value={valor}/>)
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const textoContador =  parseInt(wrapper.find('h2').text().trim())

        // console.log(textoContador)

        expect( textoContador ).toBe(valor)

    })
})