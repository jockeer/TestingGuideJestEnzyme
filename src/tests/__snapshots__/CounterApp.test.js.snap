// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <CounterApp /> debe mostrar <CounterApp /> correctamente con sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
